token/client/src/token.rs

```diff
@@ -672,20 +704,37 @@ where
     }
 
     /// Transfer tokens to another account
+    #[allow(clippy::too_many_arguments)]
     pub async fn transfer<S: Signers>(
         &self,
         source: &Pubkey,
         destination: &Pubkey,
         authority: &Pubkey,
         amount: u64,
         decimals: Option<u8>,
+        fundable_owner: Option<Pubkey>,
         signing_keypairs: &S,
     ) -> TokenResult<T::Output> {
         let signing_pubkeys = signing_keypairs.pubkeys();
         let multisig_signers = self.get_multisig_signers(authority, &signing_pubkeys);
 
-        let instructions = if let Some(decimals) = decimals {
-            [instruction::transfer_checked(
+        let mut instructions = vec![];
+
+        if let Some(recipient) = fundable_owner {
+            if *destination != self.get_associated_token_address(&recipient) {
+                return Err(TokenError::AccountInvalidAccount);
+            }
+
+            instructions.push(create_associated_token_account_idempotent(
+                &self.payer.pubkey(),
+                &recipient,
+                &self.pubkey,
+                &self.program_id,
+            ));
```

Comment:
gah, while digging through token project issues, i found something that indicated i might not be allowed to do this, because the version of the associated program on mainnet-beta lacks this instruction?

the rationale for the idempotent version here is that transfer is one of the special few functions that `must` support offline mode, so this makes it more likely to "do the right thing." eg, a user can specify they want to fund the recipient just in case, and the transaction will still work if the account exists

i also create transactions in the new version of gc that depend on this behavior because its convenient to not have to track associated account existence perfectly, but i could work around that if necessary